### PR TITLE
Introduces first() and last() methods

### PR DESCRIPTION
**Pull request by jakubgiminski**
Introduces `first()` and `last()` methods. If Collection is empty, `OutOfBoundsException` is thrown.

### CODE REVIEW

File: src/Collection.php [hunk 1]
```diff
@@ -552,4 +552,14 @@ public function merge($items)
 
         return $col;
     }
+
+    public function first()
+    {
+        return reset($this->items);
```
Comment (danielgsims) [inline src/Collection.php]: Reset and End will return false if the collection is empty, but that could be an odd response, especially if you have a list of booleans. If the internal array is empty, should there be an exception thrown?

File: src/Collection.php [hunk 2]
```diff
@@ -552,4 +552,14 @@ public function merge($items)
...
+    }
+
+    public function last()
```
Comment (danielgsims) [inline src/Collection.php]: We will want to add these methods to the CollectionInterface, too!

**Comment by danielgsims** [comment]
Thank you for the PR! These would be really handy methods. I added a couple comments, please let me know what you think. Thanks again!

File: src/Collection.php (hunk 1 above)
Comment (KracLifeworks) [inline src/Collection.php]: @jakubgiminski @danielgsims Hey you can use `reset()` + `key()` in order to achieve desired behaviour. If `key()` returns  `NULL` that means there is no element in the array.

**Comment by jakubgiminski** [comment]
@danielgsims I corrected the PR. Let me know, if there is anything else. Cheers. :)

**Comment by KracLifeworks** [comment]
@jakubgiminski You should have used `OutOfBoundsException` instead `RuntimeException`

**Comment by dkraczkowski** [comment]
@jakubgiminski @danielgsims Hey you can use `reset()` + `key()` in order to achieve desired behaviour. If `key()` returns  `NULL` that means that array is empty.
```php
$item = reset($this->items);
if (null === key($this->items)) {
    return null;
}
return $item;
```

Btw. Jakub you should have used `OutOfBoundsException` instead `RuntimeException`

**Comment by danielgsims** [comment]
👍 I dig the OutOfBoundsException. Thanks for the PR @jakubgiminski and thanks for the input @dkraczkowski

**Pull request merged**